src/util/file.c

```diff
@@ -40,6 +40,29 @@
 #include "lib/message.h"
 #include "lib/privilege.h"
 
+int has_perm(int level, struct stat file_stat) {
```

Comment:
It's not clear to me how this can be correct.  Everything is done in decimal, but permissions are done in octal.

Also not clear what permission this is testing for.  It doesn't seem to be looking for a single permission, but at least one of several permissions.  Maybe rename it to something more descriptive?